docs/DEVELOPING.md

```diff
@@ -111,9 +128,38 @@ To sync both sides, the CLI uses these steps:
   - if not, adds it as an `upsert` operation into a `bulk` call to Elasticsearch
 - for each id from Elasticsearch that is not present in the documents sent by the data source class, adds it as a `delete` operation into the `bulk` call
 - `bulk` calls are emitted every 500 operations (this is configurable for slow networks).
+- If incremental sync is available for the connector, `self._sync_cursor` should also be updated in the connector document in the end, so that when an incremental sync starts, it knows where to start from.
 
 To implement a new source, check [CONTRIBUTE.rst](./CONTRIBUTING.md)
 
+### How an incremental sync works
+
+If a connector supports incremental sync, you should firstly enable it. Check [Features](#features).
+
+You have to implement `get_docs_incrementally` method of the data source class, which will take `sync_cursor` as a parameter, and yield `document`, `lazy_download`, `operation`.
+
+```python
+async def get_docs_incrementally(self, sync_cursor, filtering=None):
+    # start syncing from sync_cursor
+    ...
+    
+    # document - a json representation of an item in the remote source
+    # lazy_download - a function to download attachment, if available
+    # operation - operation to be applied to this document, can be one of index, update or delete
```

Comment:
Probably good to mention the enum behind it and what's the difference between index and update (super important)